### ISSUE TITLE
Do not use ServiceLocator.Current in Region class

### ISSUE THREAD

**Issue by CriggerMarg** [PrismLibrary/Prism#1058]
Region.cs contains next lines

        /// <summary>
        /// Gets the navigation service.
        /// </summary>
        /// <value>The navigation service.</value>
        public IRegionNavigationService NavigationService
        {
            get
            {
                if (this.regionNavigationService == null)
                {
                    **this.regionNavigationService = ServiceLocator.Current.GetInstance<IRegionNavigationService>();**
                    this.regionNavigationService.Region = this;
                }

                return this.regionNavigationService;
            }

            set
            {
                this.regionNavigationService = value;
            }
        }

Bold line do not allow me to inject my own service locator instance into module. I trying to achieve container per module and this line is only one thing that do not allow me to do so. In all other prism classes ServiceLocator is injected with ctor and may be overidden with container registration. 

I use separate container for each module and able to override almost all dependencies. I registered my service locator instance with isolated container and use it in module. But I can't do so with ServiceLocatorProvider because I will not know where request for instance coming from. 

I suggest to inject service locator into ctor instead of using current anti-pattern.

**Comment by CriggerMarg** [comment]
I found that issue is duplicating https://github.com/PrismLibrary/Prism/issues/1042

**Comment by brianlagunas** [comment]
Let's keep this specific issue focused and not bunch it into the other general issue.  I'll have to look into this and see if this is possible.  This is the older part of the code base and I am not sure if the Region class is resolved using a container.  If that is the case then we would not be able to inject the service locator.  I'm not sure why the P&P team did it this way, but I am assuming they had their reasons.

Honestly, creating a container per modules makes me nervous and I have never seen the need for that.  I am sure you have your reasons, but this is very unorthodox.

**Comment by CriggerMarg** [comment]
Cool. I have enterprise app that has two modules inside of one project and they using same types. So I have to add prefixes for registration names during module initialize and then manipulate with this prefixes inside of dependent types. I know it's sounds intricately but it is how it is. 

If I would have separate container in module, I would be able to get rid of this prefixes and use pure type names for navigation

**Comment by CriggerMarg** [comment]
I digged into code and found that you are right - regions created directly, without DI. It makes things complicated.

**Comment by brianlagunas** [comment]
So you have an interface called IFoo that each module has a different implementation of?  So ModuleA needs FooA and ModuleB needs FooB, but both are based on IFoo?

**Comment by CriggerMarg** [comment]
Nope, it worse. I have concrete implementation that used in both modules. It's enterprise, baby. However, I think there might be solution that does not requiring changes in prism itself. I may inherit from concrete region type like SingleActiveRegion and override its NavigationService property to use injected from ctor. 
And inject this service with my region adapter. 

Maybe I violated terms of service that were written in issue placeholder but when I started I wasn't sure I can do it without prism source changes.
